war/src/main/webapp/scripts/hudson-behavior.js

```diff
@@ -1339,7 +1339,7 @@ function replaceDescription() {
  * and attached under the element identified by the specified id.
  */
 function applyNameRef(s,e,id) {
-    $(id).groupingNode = true;
```

Comment:
So I have seen the exact same random recreation of elements in [other code](https://github.com/jenkinsci/ldap-plugin/pull/18/commits/ff919c69d4dfb28962b4eb2f4ce00b5c525f23db) and my fix was to move storage to a different location. In my case I couldn't stash in an attribute because YUI was doing other changes to the dom.

My interpretation was that there is a hashmap like data structure and hence the hash-order is non-predictable leading to some of the time getting the element recreated